client/cmdhfmf.c

```diff
@@ -1530,6 +1557,423 @@ static int CmdHF14AMfNestedHard(const char *Cmd) {
+    PrintAndLogEx(NORMAL, "[+] Dumping the found keys: %s", createDumpFile ? "True" : "False");
+    PrintAndLogEx(NORMAL, "[+] Card sectors: %d", sectorsCnt);
+    PrintAndLogEx(NORMAL, "[+] Key supplied: %s", know_target_key ? "True" : "False");
+    PrintAndLogEx(NORMAL, "[+] Known sector: %d", blockNo);
+    PrintAndLogEx(NORMAL, "[+] Keytype: %c", keyType ? 'B' : 'A');
+    PrintAndLogEx(NORMAL, "[+] Kown key: 0x%02x%02x%02x%02x%02x%02x", key[0], key[1], key[2], key[3], key[4], key[5]);
+    PrintAndLogEx(NORMAL, "[+] Dictionary: %s", filename);
+
+    e_sector = calloc(sectorsCnt, sizeof(sector_t));
+
+    if (know_target_key) {
+        // check if we can authenticate to sector
+        if (mfCheckKeys(blockNo, keyType, true, 1, key, &key64) != PM3_SUCCESS) {
+            PrintAndLogEx(WARNING, "Key is wrong. Can't authenticate to sector:%3d key type:%c", blockNo, keyType ? 'B' : 'A');
+            free(e_sector);
+            return 3;
+        }
+    } else {
+        PrintAndLogEx(WARNING, "No known key was supplied, if no usable key is found in the dictionary, then this attack will fail!");
+    }
+
+    // Clear the datastructures
+    for (i=0; i<80; i++) {
+        arr[i] = 0;
+    }
+    for (i=0; i<sectorsCnt; i++) {
+        for (i2=0; i2<2; i2++) {
+            e_sector[i].Key[i2] = 0;
+            e_sector[i].foundKey[i2] = 0;
+        }
+    }
+
+    // Load the keyfiles
+    if (strlen(filename) != 0) {
+        f = fopen(filename, "r");
+        if (!f) {
+            PrintAndLogEx(FAILED, "File: " _YELLOW_("%s") ": not found or locked.", filename);
+            free(e_sector);
+            return 1;
+        }
+
+        // read file
+        while (fgets(buf, sizeof(buf), f)) {
+            if (strlen(buf) < 12 || buf[11] == '\n')
+                continue;
+            
+            while (fgetc(f) != '\n' && !feof(f)) ;  //goto next line
+
+            if (buf[0] == '#') continue; //The line start with # is comment, skip
+            if (!isxdigit(buf[0])) {
+                PrintAndLogEx(FAILED, "File content error. '" _YELLOW_("%s")"' must include 12 HEX symbols", buf);
+                continue;
+            }
+
+            buf[12] = 0;
+            if (keyitems - keycnt < 2) {
+                p = realloc(keyBlock, 6 * (keyitems += 64));
+                if (!p) {
+                    PrintAndLogEx(FAILED, "Cannot allocate memory for default keys");
+                    free(keyBlock);
+                    fclose(f);
+                    free(e_sector);
+                    return 2;
+                }
+                keyBlock = p;
+            }
+            int pos = 6 * keycnt;
+            memset(keyBlock + pos, 0, 6);
+            num_to_bytes(strtoll(buf, NULL, 16), 6, keyBlock + pos);
+            keycnt++;
+            memset(buf, 0, sizeof(buf));
+        }
+        fclose(f);
+        PrintAndLogEx(SUCCESS, "Loaded %2d keys from " _YELLOW_("%s"), keycnt, filename);
```

Comment:
try looking inside fileutils.c  there is a function to load dictionary files

Reply:
DONE